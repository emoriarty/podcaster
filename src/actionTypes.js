export const FETCH_COUNTRIES = 'itunes/FETCH_COUNTRIES';
export const FETCH_COUNTRIES_SUCCESS = 'itunes/FETCH_COUNTRIES_SUCCESS';
export const FETCH_COUNTRIES_FAIL = 'itunes/FETCH_COUNTRIES_FAIL';

export const FETCH_MEDIA_TYPES = 'itunes/FETCH_MEDIA_TYPES';
export const FETCH_MEDIA_TYPES_SUCCESS = 'itunes/FETCH_MEDIA_TYPES_SUCCESS';
export const FETCH_MEDIA_TYPES_FAIL = 'itunes/FETCH_MEDIA_TYPES_FAIL';

export const FETCH_COMMON_TRANSLATIONS = 'itunes/FETCH_COMMON_TRANSLATIONS';
export const FETCH_COMMON_TRANSLATIONS_SUCCESS = 'itunes/FETCH_COMMON_TRANSLATIONS_SUCCESS';
export const FETCH_COMMON_TRANSLATIONS_FAIL = 'itunes/FETCH_COMMON_TRANSLATIONS_FAIL';

export const FETCH_MEDIA_TYPES_TRANSLATIONS = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS';
export const FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS';
export const FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL';
