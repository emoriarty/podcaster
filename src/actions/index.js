export { default as fetchCountries } from './itunes/fetchCountries';
export { default as fetchMediaTypes } from './itunes/fetchMediaTypes';
