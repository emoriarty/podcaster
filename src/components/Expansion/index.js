export { default as FilterPanel } from './FilterPanel'
export { default as PodcastFilter } from './PodcastFilter'
