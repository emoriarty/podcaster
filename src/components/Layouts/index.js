export * from './AppContainer'
export * from './AppWrapper'
