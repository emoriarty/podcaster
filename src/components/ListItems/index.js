export { default as TwoLineListItem } from './TwoLineListItem'
export { default as TwoLineLoadingListItem } from './TwoLineLoadingListItem'
