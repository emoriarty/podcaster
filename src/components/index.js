export * from './Layouts'
export * from './ListItems'
export * from './Texts'
