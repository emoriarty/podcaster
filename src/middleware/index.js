export { default as yqlMiddleware } from './yql'
export { default as rssMiddleware } from './rss'
