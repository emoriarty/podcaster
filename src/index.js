import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import appReducer from './reducers';
import { fetchCountries } from './actions';

const loggerMiddleware = createLogger();

window.addEventListener('load', () => {
  // jsonp(
  //   'https://itunes.apple.com/search?term=carne&media=podcast',
  //   (err, response) => console.log(err, response)
  // );

  let store = createStore(
    appReducer,
    applyMiddleware(
      thunkMiddleware,
      loggerMiddleware
    )
  );

  console.log('initialState', store.getState());

  const unsubscribe = store.subscribe(() => console.log(store.getState()));

  store
    .dispatch(fetchCountries())
    .then(data => {
      console.log('success', data);
    })
    .catch(reason => {
      console.error(reason);
    });

  unsubscribe();
});
