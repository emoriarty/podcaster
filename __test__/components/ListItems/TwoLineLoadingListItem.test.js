import React from 'react'
import { TwoLineLoadingListItem } from '../../../src/components'

describe('TwoLineLoadingListItem', () => {
  it('match snapshot', () => {
    const component = shallow( <TwoLineLoadingListItem />)
    expect(component)
      .toMatchSnapshot()
  })
})
