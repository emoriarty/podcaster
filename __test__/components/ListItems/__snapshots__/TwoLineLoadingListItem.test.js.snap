// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwoLineLoadingListItem match snapshot 1`] = `
<div>
  <div>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
